packages/workspace/src/tasks-runner/task-orchestrator.ts

```diff
@@ -115,18 +118,33 @@ export class TaskOrchestrator {
     tasks.forEach((t) => {
       this.options.lifeCycle.startTask(t.task);
 
+      const outputs = getOutputs(this.projectGraph.nodes, t.task);
```

Comment:
I think you are on the right track, but you can simplify it to something like:




  private applyCachedResults(tasks: TaskWithCachedResult[]) {
    tasks.forEach((t) => {
      this.options.lifeCycle.startTask(t.task);

      const outputs = getOutputs(this.projectGraph.nodes, t.task);
      const outputsMatchCache = this.cache.outputsMatchTask(task, outputs); // this does existSync as well, and check fo for empty array

      if (! outputsMatchCache) {
        const hashes = this.cache.removeHashesFromNxOutputs(outputs);
        this.cache.copyFilesFromCache(t.cachedResult, outputs);
        this.cache.writeOutputHashesToNxOutputs(cachedOutputs, t.task.hash);
      }


      if (
        !this.initiatingProject ||
        this.initiatingProject === t.task.target.project
      ) {
        const args = this.getCommandArgs(t.task);
        output.logCommand(`nx ${args.join(' ')}`, outputsMatchCache ? MatchedExistingOutput : RetrievedFromCache);
        process.stdout.write(t.cachedResult.terminalOutput);
      }

      this.options.lifeCycle.endTask(t.task, 0);
    });
    
    
    Not reassignments, fewer if statements.

Reply:
Thanks for the suggestion! I'll re-work it to simplify it, it's definitely over complicated atm 😅 
I think is still worthy to have the `cachedOutputs` and work with that instead of the full `outputs` because we can have in that `outputs` array values that were not used by the build, therefore they don't really exist and we pollute the `nx-outputs.json` with them. What do you think?